app/controllers/catalog_controller.rb

```diff
@@ -1943,11 +1943,7 @@ def replace_right_cell(action = nil, replace_trees = [])
     presenter[:reload_toolbars][:view]    = v_tb
     presenter[:set_visible_elements][:toolbar] = h_tb.present? || c_tb.present? || v_tb.present?
 
-    if @record && !@in_a_form
-      presenter[:record_id] = @record.id
-    else
-      presenter[:record_id] = @edit && @edit[:rec_id] && @in_a_form ? @edit[:rec_id] : nil
-    end
+    set_presenter_record_id(presenter)
```

Comment:
@Fryguy @matthewd is it better/clearer to have something like the following instead:
```ruby
presenter[:record_id] = determine_record_id_for_presenter
```